src/create-context.js

```diff
@@ -21,18 +21,24 @@ export function createContext(defaultValue) {
 
 				this.shouldComponentUpdate = _props => {
 					if (this.props.value !== _props.value) {
-						subs.some(c => {
-							c.context = _props.value;
-							enqueueRender(c);
+						subs.some(payload => {
+							payload[0].context = _props.value;
+							if (
+								!payload[1] ||
+								payload[1]._update(_props.value, this.props.value)
+							) {
+								enqueueRender(payload[0]);
```

Comment:
if a component reads from `MyContext1` and then in later render reads from `MyContext2` the sub for `MyContext1` will hang around and you will run the updater meant for `MyContext2` whenever `MyContext1` provides a new value, possibly leading to errors or unnecessary render queuing.

This is actually already true, minus the updater, b/c even without the updater the sub isn't cleaned up until unmount it seems. so the part that is new is surprising errors running a shouldUpdate intended for a new context using the old contexts values

Reply:
Is this allowed? Hmm TIL, will make a quick PR for that 👍

Reply:
yes, I don't think users often take advantage of it but the React API does not prohibit it. their model encodes the dependency on a context on the dependent component not on the provider presumably for this very reason. tracking which components need to be notified in a provider is hard when that dependency may not be uniform over time. without selectors it is mostly just a deop where you render some things more than needed but with the selector (or updater in this case) you introduce potential for runtime errors so more consequential. I will say that with 2 phase render React has a much harder job doing to the dependency tracking and with a 1 phase it seems much more possible so I wouldn't necessarily adopt their approach